stdlib/src/utils/string_slice.mojo

```diff
@@ -144,10 +171,11 @@ struct _StringSliceIter[
         self.index = 0 if forward else length
         self.ptr = unsafe_pointer
         self.length = length
-        self.continuation_bytes = 0
-        for i in range(length):
-            if _utf8_byte_type(unsafe_pointer[i]) == 1:
-                self.continuation_bytes += 1
+        self.continuation_bytes = _count_utf8_continuation_bytes(
```

Comment:
I think you can write `_count_utf8_continuation_bytes(self._slice)`?

Reply:
I could change the input from ptr and len to span but it just gives the work to all places where a `_StringSliceIter` is instantiated, and I tried and it ends up with an ugly lifetime mess for something that is internal and should be simple. And this PR would become bigger.

Reply:
I'm fine as is! Plus we can always land and iterate.